pkg/util/goroutinemap/common/exponential_backoff.go

```diff
@@ -0,0 +1,77 @@
+/*
```

Comment:
Thats fine, just wanted to make sure you were aware.

On Tue, Jul 19, 2016 at 5:42 PM, Saad Ali <notifications@github.com> wrote:

> In pkg/util/goroutinemap/common/exponential_backoff.go
> <https://github.com/kubernetes/kubernetes/pull/28939#discussion_r71429500>
> :
>
> > @@ -0,0 +1,77 @@
> > +/*
>
> This logic is pretty specific to the way that goroutinemap operates. The
> package clayton mentioned, pkg/util/wait/ExponentialBackoff, is a more
> generic mechanism for doing exponential backoff and is already in pkg/util.
> That package takes responsibility for execution (repeatedly execute
> function with exp backoff until it succeeds). Whereas goroutinemap expects
> the caller to control function execution and simply blocks it if it is too
> frequent. Combining the two is a non-trivial task and outside the scope of
> this PR. And because this backoff mechanism is unique to the way that
> goroutinemap operates I'd prefer to leave it here until a more through
> refactor can be done combining the two.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28939/files/f344f35d4d1b693130ed5f82f76696c9064b2609#r71429500>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p42yl4O1PM82jU1Llk8BMBpcluE7ks5qXUTKgaJpZM4JMGj8>
> .
>